internal/controllers/provisioningrequest_controller_test.go

```diff
@@ -516,8 +516,11 @@ var _ = Describe("ProvisioningRequestReconcile", func() {
 					utils.HwTemplateNodePool: `
     - name: master
       hwProfile: profile-spr-single-processor-64G
+      role: master
```

Comment:
Use `controller` for the name here for testing, so that it's different than the role. And include a resourcePoolId